webapps/frontend/ui/admin/client/scripts/pages/execution-metrics.js

```diff
@@ -17,190 +17,358 @@
+const debounceAnnual = debouncePromiseFactory();
 
-var Controller = [
+const fmtMonth = 'MMMM YYYY';
+const fmtYear = 'DD/MM/YYYY';
+const fmtDatePicker = 'yyyy-MM-dd';
+const fmtRequest = 'YYYY-MM-DD';
+const metrics = {
+  PI: 'process-instances',
+  DI: 'decision-instances',
+  TU: 'task-users',
+  FNI: 'flow-node-instances',
+  EDE: 'executed-decision-elements'
+};
+const localConfContractStartDate = 'metricsContractStartDate';
+
+const Controller = [
   '$scope',
   '$filter',
-  'Uri',
   'camAPI',
-  'fixDate',
   'configuration',
-  function($scope, $filter, Uri, camAPI, fixDate, configuration) {
-    var MetricsResource = camAPI.resource('metrics');
-
-    // date variables
-    var now = new Date();
-    var dateFilter = $filter('date');
-    var dateFormat = "yyyy-MM-dd'T'HH:mm:ss";
+  'localConf',
+  'PluginMetricsResource',
+  '$translate',
+  function(
+    $scope,
+    $filter,
+    camAPI,
+    configuration,
+    localConf,
+    PluginMetricsResource,
+    $translate
+  ) {
+    const telemetryResource = camAPI.resource('telemetry');
+    const dateFilter = $filter('date');
 
     // initial scope data
-    $scope.startDate = dateFilter(
-      now.getFullYear() + '-01-01T00:00:00.000',
-      dateFormat
-    );
-    $scope.endDate = dateFilter(
-      now.getFullYear() + '-12-31T23:59:59.999',
-      dateFormat
-    );
+    $scope.fmtDatePicker = fmtDatePicker;
+    let startDate = localConf.get(localConfContractStartDate);
+    if (startDate) {
+      $scope.startDate = startDate;
+    } else {
+      startDate = moment()
+        .startOf('year')
+        .toDate();
+    }
+    $scope.startDate = dateFilter(startDate, fmtDatePicker);
+
     $scope.loadingState = 'INITIAL';
-    $scope.alwaysShowUTWMetrics = configuration.getAlwaysShowUniqueTaskWorkerMetrics();
-    $scope.showTaskWorkerMetric = $scope.alwaysShowUTWMetrics;
+    $scope.loadingStateMonthly = 'INITIAL';
+    $scope.loadingStateAnnual = 'INITIAL';
     $scope.metrics = {};
 
+    let monthlyMetricUsageMap = {};
+    let monthlyMetricsArray = {};
+    $scope.monthlyMetrics = [];
+    $scope.annualMetrics = [];
+    $scope.displayLegacyMetrics = false;
+    $scope.datePickerOptions = {maxDate: moment().toDate()};
+
+    telemetryResource.fetchData((err, res) => {
+      if (!err) {
+        $scope.telemetryData = res;
+        delete $scope.telemetryData.product.internals.commands;
+        delete $scope.telemetryData.product.internals.metrics;
+      } else {
+        $scope.telemetryData = $translate.instant(
+          'DIAGNOSTICS_FETCH_DATA_ERROR_MESSAGE',
+          {
+            err
+          }
+        );
+      }
+    });
+
     // sets loading state to error and updates error message
-    function setLoadingError(error) {
-      $scope.loadingState = 'ERROR';
-      $scope.loadingError = error;
-    }
+    const setLoadingError = error => {
+      $scope.loadingStateMonthly = $scope.loadingStateAnnual = 'ERROR';
```

Comment:
🙃 I would find this confusing, when reading the code quickly. I would prefer:
```
$scope.loadingStateMonthly = 'ERROR';
$scope.loadingStateAnnual = ERROR';
$scope.loadingErrorMonthly = error;
$scope.loadingErrorAnnual = error;
```

Reply:
I think I'll keep this one, I find it neat 😀